File: workbench-server/src/main/java/org/example/workbenchserver/entity/Memo.java
package org.example.workbenchserver.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;

import java.time.LocalDateTime;

/**
 * 备忘录实体，对应 {@code wb_memo}（docs/数据模型.md §4.4）。
 *
 * <p><b>本类刻意没有 {@code userId} 字段，这不是漏写</b>，理由同
 * {@link PlanTask}：租户插件只在 INSERT 时补列清单里没有的列，
 * 实体一旦带 {@code userId} 就成了绕过隔离的越权写入通道。
 *
 * <p>{@code content} 可为空 —— 建表脚本里它就是 NULL 列。空正文由
 * Service 归一成空串而非保留 null，理由见 {@code MemoServiceImpl.apply}。
 */
@TableName("wb_memo")
public class Memo {

	@TableId(type = IdType.AUTO)
	private Long id;

	/** 标题。没写标题时存空串，列表会退而显示正文摘要 */
	private String title;

	/** 正文。TEXT 列，可长可空 */
	private String content;

	private LocalDateTime createTime;

	private LocalDateTime updateTime;

	/** 逻辑删除：0 未删除，非 0 为删除时间戳 */
	@TableLogic
	private Long deleted;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public String getContent() {
		return content;
	}

	public void setContent(String content) {
		this.content = content;
	}

	public LocalDateTime getCreateTime() {
		return createTime;
	}

	public void setCreateTime(LocalDateTime createTime) {
		this.createTime = createTime;
	}

	public LocalDateTime getUpdateTime() {
		return updateTime;
	}

	public void setUpdateTime(LocalDateTime updateTime) {
		this.updateTime = updateTime;
	}

	public Long getDeleted() {
		return deleted;
	}

	public void setDeleted(Long deleted) {
		this.deleted = deleted;
	}

}

File: stfworkbench-vue/src/types/memo.ts
/**
 * 备忘录相关类型，与后端 MemoVO / MemoDetailVO 一一对应，
 * 见 docs/接口清单.md §6。
 */

/**
 * 列表项。**不含正文** —— 接口刻意只给一段摘要。
 *
 * 正文可能上万字，列表页一条也用不上，全查出来只是白白多传几 MB。
 * 要看正文走详情接口（{@link MemoDetail}）。
 */
export interface Memo {
  id: number
  /** 可能为空串：后端只要求"标题和正文不能都为空"，标题单独可以空 */
  title: string
  /** 正文压成一行后的前 60 个字，**由后端截好**，前端不要再截一次 */
  summary: string
  /** yyyy-MM-dd HH:mm:ss */
  updateTime: string
}

/** 详情。比列表项多出正文和创建时间 */
export interface MemoDetail {
  id: number
  title: string
  content: string
  createTime: string
  updateTime: string
}

/**
 * 新增 / 修改的入参。
 *
 * 两者共用一份：后端的 PUT 是全量替换，字段集合与 POST 完全一致。
 * 想清空就传空串 —— 后端把 null 和空值一律落成空串再写库
 * （见 MemoServiceImpl#apply），不存在"不传即保留"。
 */
export interface MemoParams {
  title: string
  content: string
}
